### DISCUSSION TITLE
Issue in using Image Widget

### DISCUSSION THREAD

**Discussion by tareksalem** [hecrj/iced#1245] [Q&A] [answered]
I am just new to rust, I am trying to use `Image` widget as described in the documentation as the following:

```rust
Image::new("https://raw.githubusercontent.com/iced-rs/iced/9712b319bb7a32848001b96bd84977430f14b623/examples/resources/ferris.png")
```
however it gives me the following error while compilation process

```
the trait bound `iced_wgpu::backend::Backend: iced_graphics::backend::Image` is not satisfied
  --> src/main.rs:40:13
   |
39 | ...et body = Column::new().push(render_mark_components()).width(Length::FillPortion(4)).push(
   |                                                                                         ---- required by a bound introduced by this call
40 | ...   Image::new("https://raw.githubusercontent.com/iced-rs/iced/9712b319bb7a32848001b96bd84977430f14b623/examples/resources/ferris.png").height(Length::Units(20)).width(Length::Units(10))
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `iced_graphics::backend::Image` is not implemented for `iced_wgpu::backend::Backend`
   |
   = note: required because of the requirements on the impl of `iced_native::widget::image::Renderer` for `iced_graphics::renderer::Renderer<iced_wgpu::backend::Backend>`
   = note: required because of the requirements on the impl of `From<Image>` for `iced_native::element::Element<'_, Message, iced_graphics::renderer::Renderer<iced_wgpu::backend::Backend>>`
   = note: required because of the requirements on the impl of `Into<iced_native::element::Element<'_, Message, iced_graphics::renderer::Renderer<iced_wgpu::backend::Backend>>>` for `Image`
note: required by a bound in `iced_native::widget::column::Column::<'a, Message, Renderer>::push`
  --> /home/tarek/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_native-0.4.0/src/widget/column.rs:97:12
```

**Comment by Kaiden42** [comment, marked as answer]
You have to enable the `image` feature of iced.

For example:
```
iced = { git = "https://github.com/hecrj/iced.git", features = ["image"]}
```

**Reply by tareksalem** [reply]
thanks for help, I updated it and the error is gone however it doesn't output the image

**Reply by tareksalem** [reply]
![Screenshot from 2022-02-10 02-53-33](https://user-images.githubusercontent.com/26143873/153315972-465ab294-4cc6-4e7d-953a-1efbad49437d.png)
as you can see the image place is blank

**Reply by Kaiden42** [reply]
That's because iced is not a web framework (unless you are building for web) therefore your image needs to be located on your machine and not the internet.

(You could download it into memory and build an image handle from the bytes in the memory like [here](https://github.com/iced-rs/iced/blob/adce9e04213803bd775538efddf6e7908d1c605e/examples/pokedex/src/main.rs#L208), but I wouldn't recommend to do this as a beginner.)

**Reply by tareksalem** [reply]
Actually what I want to do is grap some articles from api and get the link of image related to each article and show the image,

**Reply by Kaiden42** [reply]
Then I would recommend you to take a look at the full [pokemon example](https://github.com/iced-rs/iced/tree/master/examples/pokedex) to learn how to access things from the internet.

**Reply by tareksalem** [reply]
thanks bro
